### PR TITLE
Guidelines: Add the entry about one-off scripts

### PR DESCRIPTION
**Pull request by evgenyz**
#### Description:

- We don't want one-off stray scripts to lay around.

#### Rationale:

- They litter the project space. In rare cases when a script could be useful again,
it could be retrieved from the history.

### CODE REVIEW

File: docs/manual/developer/04_style_guide.md [hunk 1]
```diff
@@ -46,20 +46,24 @@ In practice, this implies:
...
-* Don't put authorship information into the code, Git tracks authorship for you.
-  Don't copy-paste license text into source files -- use [SPDX IDs](https://spdx.dev/ids/) for that purpose.
-* Don't take part in making files longer - files longer than 400 lines should be an exception.
-  Add your new code into a new file, and possibly move existing code to it in the same or in a follow-up PR.
 
 When working with existing code that doesn't satisfy these recommendations, simply leave the code in a better shape than the shape in which it was before,
 and keep these guidelines in mind when writing new code.
 
+## Project-level Coding Style
+* Include tests for your contribution.
```
Comment (ggbecker) [inline docs/manual/developer/04_style_guide.md:57]:
```suggestion
## Project-level Coding Style

* Include tests for your contribution.
```

codeclimate uncovered issue

File: docs/manual/developer/04_style_guide.md [hunk 2]
```diff
@@ -46,20 +46,24 @@ In practice, this implies:
...
+* Don't take part in making files longer -- files longer than 400 lines should be an exception.
+  Add your new code into a new file, and possibly move existing code to it in the same or in a follow-up PR.
+* Don't copy-paste code, use e.g. Jinja macros to reduce duplication.
+  Exception to this rule is code that is identical another piece of code only by coincidence, and there is a substantial probability that the code can diverge.
+* Don't put authorship information into the code, Git tracks authorship for you.
+  Don't copy-paste license text into source files -- use [SPDX IDs](https://spdx.dev/ids/) for that purpose.
+* Don't commit one-off scripts to the project.
```
Comment (ggbecker) [inline docs/manual/developer/04_style_guide.md:64]:
```suggestion
* Don't commit one-off scripts to the project.
```
codeclimate uncovered issue

File: docs/manual/developer/04_style_guide.md [hunk 3]
```diff
@@ -46,20 +46,24 @@ In practice, this implies:
...
+* Include tests for your contribution.
+* Don't take part in making files longer -- files longer than 400 lines should be an exception.
+  Add your new code into a new file, and possibly move existing code to it in the same or in a follow-up PR.
+* Don't copy-paste code, use e.g. Jinja macros to reduce duplication.
+  Exception to this rule is code that is identical another piece of code only by coincidence, and there is a substantial probability that the code can diverge.
+* Don't put authorship information into the code, Git tracks authorship for you.
+  Don't copy-paste license text into source files -- use [SPDX IDs](https://spdx.dev/ids/) for that purpose.
+* Don't commit one-off scripts to the project. 
+  On occasions when one would like to get a feedback on the approach or debug the script,
+  that makes automated changes to the project in a PR, it is strongly recommended to add a removal commit before merging the PR.
```
Comment (ggbecker) [inline docs/manual/developer/04_style_guide.md:66]: codeclimate also complains about line length but it shouldn't be a deal breaker in this case, if you are in the mood of fixing it then you can do it :)

**Review by ggbecker — COMMENTED** [review]
We could find such scripts and delete them from the project as well. For example, the csv to rule migration tool is obsolete and if anyone would still require it, they can obtain from an older release.

https://github.com/ComplianceAsCode/content/blob/master/utils/migrate_template_csv_to_rule.py

File: docs/manual/developer/04_style_guide.md (hunk 3 above)
Comment (matejak) [reply docs/manual/developer/04_style_guide.md:66]: I would prioritize the single logical statement per line even if the line becomes somehow longer.
If you are in the mood, feel free to turn off md line length checker by means of the `.mdlrc` file. I think that other issues detected are somewhat valuable, so I wouldn't turn the whole md checker off.

File: docs/manual/developer/04_style_guide.md (hunk 2 above)
Comment (evgenyz) [reply docs/manual/developer/04_style_guide.md]: Fixed

File: docs/manual/developer/04_style_guide.md (hunk 1 above)
Comment (evgenyz) [reply docs/manual/developer/04_style_guide.md]: Fixed

File: docs/manual/developer/04_style_guide.md (hunk 3 above)
Comment (evgenyz) [reply docs/manual/developer/04_style_guide.md:70]: Length checker is off.

**Comment by evgenyz** [comment]
The more I'm trying to fix mdl errors, the more errors I get (even from unrelated files). I'd say that it is good enough, since there is no more problems in the Style Guilde file.

**Comment by matejak** [comment]
> The more I'm trying to fix mdl errors, the more errors I get (even from unrelated files). I'd say that it is good enough, since there is no more problems in the Style Guilde file.

I agree, thank you for the PR! Could you please just rebase, so the failing test starts to pass as well?

**Review by matejak — APPROVED** [review]

**Comment by matejak** [comment]
Thanks, fail on rawhide during dependency install is OK.

**Pull request merged**